doc/rfc-move-sigverify-out-of-mask-rom.md

```diff
@@ -0,0 +1,80 @@
+# RFC: Move sigverify out of the mask ROM
+
+Author: James Wainwright
+Date: 2025-04-17
+Status: draft
+
+## Summary
+
+This RFC proposes moving the silicon creator signature verification code from the mask ROM to the immutable ROM extension, which would no longer be optional nor signed.
+
+The mutable ROM extension would be renamed to the FMC (first mutable code), and the term ROM extension would refer only to what we currently call the immutable ROM extension.
+
+## Motivation
+
+Moving sigverify out of the mask ROM enables:
+
+* Fixing and updating the cryptography code post tape-out.
+* Using different signature schemes for different SKUs of the same tape-out.
+
+Signature verification is relatively complex code and removing it from the mask derisks tape-outs.
+
+Allowing for personalised signature schemes increases the utility and value of a tape-out and potentially reduces costs if multiple customers can benefit. It also allows for SKUs which don't use signature verification and are closer to a generic microcontroller.
+
+Experimental and new signature schemes (for example using post-quantum cryptography) can be tested on a subset of personalised silicon without needing to be included in all chips.
+
+Simultaneously, making the immutable ROM extension non-optional simplifies the boot flow by keeping it linear.
+
+## Explanation
+
+The current boot process for Earlgrey at a high level is:
+
+1. Mask ROM boots.
+2. Signature verification of whole ROM extension.
+3. (optional) Hash check of, then jump to, immutable ROM extension.
+4. Jump to mutable ROM extension.
+5. Signature verification of, then jump to, owner code.
+
+The immutable ROM extension is currently a subset of the whole ROM extension and included in the silicon creator signature. Its hash is programmed to OTP during personalisation.
+
+The proposed change to the boot process would be:
+
+1. Mask ROM boots.
+2. Hash check of, then jump to, immutable ROM extension.
```

Comment:
I'd like to see a portion of flash dedicated to holding the immutable section.  The current scheme involving verification of the FMC (nee ROM_EXT) first, possibly dealing with address translation and then jumping to the immutable section has been somewhat troublesome.

It would be nice to have a segment of flash dedicated to the immutable section that is not part of the main flash data payload.  This may complicate the hardware design and selection/integration of the flash macro.